app/Notification.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import './_ciq-notification.scss';
+
+class Notification extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { messages: [] };
+        props.notifier.onMessage(this.onMessage);
```

Comment:
just a nitpick: you can use destructuring for this 

    const { notifier } = this.props;